addons/website_sale/static/src/js/website_sale_payment.js

```diff
@@ -1,124 +1,46 @@
 /** @odoo-module **/
 
-    import { debounce } from "@web/core/utils/timing";
 import checkoutForm from "payment.checkout_form";
-    import publicWidget from "web.public.widget";
-
-    const websiteSalePaymentMixin = {
-
-        /**
-         * @override
-         */
-        init: function () {
-            this._onClickTCCheckbox = debounce(this._onClickTCCheckbox, 100, true);
-            this._super(...arguments);
-        },
-
-        /**
-         * @override
-         */
-        start: function () {
-            this.$checkbox = this.$('#checkbox_tc');
-            this.$submitButton = this.$('button[name="o_payment_submit_button"]');
-            this._adaptConfirmButton();
-            return this._super(...arguments);
-        },
-
-        //--------------------------------------------------------------------------
-        // Private
-        //--------------------------------------------------------------------------
-
-        /**
-         * Update the data on the submit button with the status of the Terms and Conditions input.
-         *
-         * @private
-         * @return {undefined}
-         */
-        _adaptConfirmButton: function () {
-            if (this.$checkbox.length > 0) {
-                const disabledReasons = this.$submitButton.data('disabled_reasons') || {};
-                disabledReasons.tc = !this.$checkbox.prop('checked');
-                this.$submitButton.data('disabled_reasons', disabledReasons);
-            }
-        },
-
-    };
-
-    checkoutForm.include(Object.assign({}, websiteSalePaymentMixin, {
-        events: Object.assign({}, checkoutForm.prototype.events, {
+import publicWidget from "web.public.widget";
+import core from "../../../../web/static/src/legacy/js/services/core";
+
+const websiteSalePaymentMixin = {
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.$checkbox = this.$('#checkbox_tc');
+        this.$submitButton = this.$('button[name="o_payment_submit_button"]');
+        return this._super(...arguments);
+    },
+
+    /**
+     * Trigger _enableButton of payment button widget when terms and conditions checked.
+     *
+     * @private
+     * @return {undefined}
+     */
+    _onClickTCCheckbox: function () {
+        if (this.$checkbox.prop('checked')) {
+            core.bus.trigger('enableButton');
+        } else {
+            this.$submitButton.attr('disabled', true);
+        }
+    },
+}
+
+checkoutForm.include(Object.assign({}, websiteSalePaymentMixin, {
+    events: Object.assign({}, checkoutForm.prototype.events, {
+        'change #checkbox_tc': '_onClickTCCheckbox',
+    }),
+}));
+
+publicWidget.registry.WebsiteSalePayment = publicWidget.Widget.extend(
+    Object.assign({}, websiteSalePaymentMixin, {
+        selector: 'div[name="o_website_sale_free_cart"]',
+        events: {
             'change #checkbox_tc': '_onClickTCCheckbox',
```

Comment:
This is already much better, but it could be even simpler. Make all this a simple all-in-one "Terms and Conditions" widget that attaches itself to the checkbox. If it's there, the widget is loaded, otherwise it's not.